aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -368,9 +364,18 @@ Tensor empty_like(
 Tensor new_empty(
     const Tensor& self,
     IntArrayRef size,
-    const TensorOptions& options
+    c10::optional<ScalarType> dtype_opt,
+    c10::optional<Layout> layout_opt,
+    c10::optional<Device> device_opt,
+    c10::optional<bool> pin_memory_opt
     ) {
-  return at::empty(size, self.options().merge_in(options));
+  return at::empty(size,
+    dtype_opt.value_or_else([&] {return typeMetaToScalarType(self.options().dtype());}),
+    layout_opt.value_or(self.options().layout()),
+    device_opt.value_or(self.options().device()),
```

Comment:
NB: this is only semantics preserving because `self.options()` is guaranteed to have settings for dtype, layout and device. If it were possible for these to be nullopt then this transformation would not be semantics preserving

Reply:
good point. Do you think I can keep relying on it or should I rewrite this to use `self.options().layout_opt()`?

Reply:
nvm, I rewrote it